com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs

```diff
@@ -205,81 +393,123 @@ bool StripUnusedFeatures(ShaderFeatures features, Shader shader, ShaderSnippetDa
+                    !IsFeatureEnabled(features, ShaderFeatures.ScreenSpaceShadows);
+                if (mainLightShadows && !compilerData.shaderKeywordSet.IsEnabled(m_CastingPunctualLightShadow))
+                    return true;
+            }
 
             // Additional light are shaded per-vertex or per-pixel.
-            bool isFeaturePerPixelLightingEnabled = IsFeatureEnabled(features, ShaderFeatures.AdditionalLights);
-            bool isFeaturePerVertexLightingEnabled = IsFeatureEnabled(features, ShaderFeatures.VertexLighting);
-            bool clusteredRendering = IsFeatureEnabled(features, ShaderFeatures.ClusteredRendering);
-            bool isAdditionalLightPerPixel = compilerData.shaderKeywordSet.IsEnabled(m_AdditionalLightsPixel);
-            bool isAdditionalLightPerVertex = compilerData.shaderKeywordSet.IsEnabled(m_AdditionalLightsVertex);
+            // TODO: Strip off variants once we support no additional lights re-used variants
+            if (stripTool.StripMultiCompileKeepOffVariant(m_AdditionalLightsVertex, ShaderFeatures.VertexLighting,
+                m_AdditionalLightsPixel, ShaderFeatures.AdditionalLights))
+                return true;
 
-            // Strip if Per-Pixel lighting is NOT used in the project and the
-            // Per-Pixel (_ADDITIONAL_LIGHTS) variant is enabled in the shader.
-            if (!isFeaturePerPixelLightingEnabled && isAdditionalLightPerPixel)
+            if (stripTool.StripMultiCompile(m_ClusteredRendering, ShaderFeatures.ClusteredRendering))
                 return true;
 
-            // Strip if Per-Vertex lighting is NOT used in the project and the
-            // Per-Vertex (_ADDITIONAL_LIGHTS_VERTEX) variant is enabled in the shader.
-            if (!isFeaturePerVertexLightingEnabled && isAdditionalLightPerVertex)
+            // Screen Space Occlusion
+            if (stripTool.StripMultiCompile(m_ScreenSpaceOcclusion, ShaderFeatures.ScreenSpaceOcclusion))
                 return true;
 
-            if (!clusteredRendering && compilerData.shaderKeywordSet.IsEnabled(m_ClusteredRendering))
+            // Decal DBuffer
+            if (stripTool.StripMultiCompile(
+                m_DBufferMRT1, ShaderFeatures.DBufferMRT1,
+                m_DBufferMRT2, ShaderFeatures.DBufferMRT2,
+                m_DBufferMRT3, ShaderFeatures.DBufferMRT3))
                 return true;
 
-            // Screen Space Shadows
-            if (!IsFeatureEnabled(features, ShaderFeatures.ScreenSpaceShadows) &&
-                compilerData.shaderKeywordSet.IsEnabled(m_MainLightShadowsScreen))
+            // TODO: Test against lightMode tag instead.
+            if (snippetData.passName == kPassNameGBuffer)
+            {
+                if (!IsFeatureEnabled(features, ShaderFeatures.DeferredShading))
+                    return true;
+            }
+
+            // TODO: make sure vulkan works after refactor
```

Comment:
Has this been checked? If not I assume that will be part of the QA testing so we can remove this comment.